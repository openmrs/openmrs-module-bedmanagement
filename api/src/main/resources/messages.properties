${project.parent.artifactId}.title=bedmanagement Module
${project.parent.artifactId}.admissionLocations=Admission Locations
${project.parent.artifactId}.bedTypes=Bed Types
${project.parent.artifactId}.bedTags=Bed Tags