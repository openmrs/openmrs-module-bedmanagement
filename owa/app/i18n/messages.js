export default {
    en: {
        ADMISSION_LOCATIONS: 'Admission Locations',
        BED_TYPES: 'Bed Types',
        BED_TAGS: 'Bed Tags',
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language',
        ROW_REQUIRED_MSG: 'Row is required field',
        COLUMN_REQUIRED_MSG: 'Column is required field',
        ROW_SHOULD_GREATER_THAN_ZERO: 'Row value should be greater than 0',
        ROW_SHOULD_GREATER_THAN_LAYOUT_ROW: 'Row value should not be greater than layout row size',
        COLUMN_SHOULD_GREATER_THAN_ZERO: 'Column value should be greater than 0 and less than 11',
        COLUMN_SHOULD_GREATER_THAN_LAYOUT_COLUMN: 'Row value should not be greater than layout row size',
        FIX_ERROR_MSG: 'Fix error before submit',
        ADMISSION_LOCATION_SAVE_MSG: 'Admission location save successfully',
        BED_LAYOUT_SAVE_MSG: 'Admission location bed layout save successfully',
        BED_TAG_SAVE_MSG: 'Bed Tag save successfully',
        BED_TYPE_SAVE_MSG: 'Bed Type save successfully',
        BED_SAVE_MSG: 'Bed save successfully',
        DELETE_ADMISSION_LOCATION_CONFIRM_MESSAGE:
            'Are you sure you want to delete admission location {location_name}?',
        DELETE_BED_CONFIRM_MESSAGE: 'Are you sure you want to delete bed number {bed_number}?',
        DELETE_BED_TAG_CONFIRM_MSG: 'Are you sure you want to delete bed Tag {bed_tag_name}?',
        DELETE_BED_TYPE_CONFIRM_MSG: 'Are you sure you want to delete bed type {bed_type_name}?',
        DELETE_SUCCESSFULLY: 'Delete successfully',
        SUCCESS: 'Success',
        ERROR: 'Error',
        INFO: 'Info',
        SAVE: 'Save',
        SAVING: 'Saving...',
        CANCEL: 'Cancel',
        ADD: 'Add',
        EDIT: 'Edit',
        DELETE: 'Delete',
        ADD_BED: 'Add Bed',
        ROWS: 'Rows',
        ROW: 'Row',
        COLUMNS: 'Columns',
        COLUMN: 'Column',
        SET_LAYOUT: 'Set Layout',
        LOCATION: 'Location',
        BED: 'Bed',
        BED_NUMBER: 'Bed Number',
        BED_TYPE: 'Bed Type',
        WARD: 'Ward',
        PARENT_LOCATION: 'Parent Location',
        NONE: 'None',
        NAME: 'Name',
        DESCRIPTION: 'Description',
        DISPLAY_NAME: 'Display Name',
        ACTION: 'Action',
        ADD_NEW: 'Add New',
        BED_TAG: 'Bed Tag',
        EXISTING_BED_TYPES: 'Existing Bed Types',
        EXISTING_BED_TAGS: 'Existing Bed Tags'
    }
};
