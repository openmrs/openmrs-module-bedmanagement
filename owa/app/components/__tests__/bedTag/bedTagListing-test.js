import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {IntlProvider} from 'react-intl';

import BedTagList from 'components/bedTag/bedTagList';
import bedTagFunctionsMock from 'components/__mocks__/bedTagFunctions-mock';
import messages from 'i18n/messages';

const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    props: {
        bedTagFunctions: bedTagFunctionsMock,
        bedTags: bedTagFunctionsMock.getBedTags()
    },
    context: {
        intl: intl
    }
};

describe('BedTagList', () => {
    it('Should render bed tag list properly', () => {
        const bedTagList = shallow(
            <BedTagList bedTags={testData.props.bedTags} bedTagFunctions={testData.props.bedTagFunctions} />,
            {context: testData.context}
        );

        expect(bedTagList.find('BedTagListRow').length).toBe(4);
        expect(shallowToJson(bedTagList)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnAddNewHandler = jest.spyOn(BedTagList.prototype, 'addNewHandler');
        const spyOnSetState = jest.spyOn(testData.props.bedTagFunctions, 'setState');
        const bedTagList = mount(
            <BedTagList bedTags={testData.props.bedTags} bedTagFunctions={testData.props.bedTagFunctions} />,
            {context: testData.context}
        );

        bedTagList.find("button[value='Add New']").simulate('click');
        expect(spyOnAddNewHandler).toHaveBeenCalled();
        expect(spyOnSetState).toHaveBeenCalledWith({
            activePage: 'addEdit',
            pageData: {
                operation: 'add',
                bedTagId: null
            }
        });
    });
});
