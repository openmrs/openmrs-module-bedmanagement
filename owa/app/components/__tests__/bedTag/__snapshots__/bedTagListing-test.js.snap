// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedTagList Should render bed tag list properly 1`] = `
<fieldset
  className="bed-Tag-listing"
>
  <legend>
      
    Existing Bed Tags
      
  </legend>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th
          className="description"
        >
          Description
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <BedTagListRow
        bedTag={
          Object {
            "id": 2,
            "name": "Broken",
            "uuid": "e67ce6b1-8382-4514-80db-6d67a6611534",
          }
        }
        bedTagFunctions={
          Object {
            "fetchBedTags": [Function],
            "getBedTagByUuid": [Function],
            "getBedTags": [Function],
            "notify": [Function],
            "setState": [Function],
          }
        }
        key="0"
      />
      <BedTagListRow
        bedTag={
          Object {
            "id": 3,
            "name": "Oxygen",
            "uuid": "25d9b2c8-7b70-48a5-a033-1051dcf7cc8a",
          }
        }
        bedTagFunctions={
          Object {
            "fetchBedTags": [Function],
            "getBedTagByUuid": [Function],
            "getBedTags": [Function],
            "notify": [Function],
            "setState": [Function],
          }
        }
        key="1"
      />
      <BedTagListRow
        bedTag={
          Object {
            "id": 4,
            "name": "Isolation",
            "uuid": "ff7ed494-7b9c-4478-812a-5187e297f94c",
          }
        }
        bedTagFunctions={
          Object {
            "fetchBedTags": [Function],
            "getBedTagByUuid": [Function],
            "getBedTags": [Function],
            "notify": [Function],
            "setState": [Function],
          }
        }
        key="2"
      />
      <BedTagListRow
        bedTag={
          Object {
            "id": 5,
            "name": "Sanitation required",
            "uuid": "39376170-c020-4e19-9960-c6f9044886fd",
          }
        }
        bedTagFunctions={
          Object {
            "fetchBedTags": [Function],
            "getBedTagByUuid": [Function],
            "getBedTags": [Function],
            "notify": [Function],
            "setState": [Function],
          }
        }
        key="3"
      />
    </tbody>
  </table>
  <button
    className="list-btn"
    onClick={[Function]}
    value="Add New"
  >
    Add New
  </button>
</fieldset>
`;
